src/librustc_trans/trans/mir/operand.rs

```diff
@@ -94,6 +94,24 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
 
         match *operand {
             mir::Operand::Consume(ref lvalue) => {
+                // watch out for temporaries that do not have an
```

Comment:
I wonder if this "trans operand into" path is all that useful. It seems like we could generate the same code by just calling `trans_operand` and then store the resulting operand instead, so as not to duplicate this logic.